### PR TITLE
Host language debugger

### PR DESCRIPTION
**Pull request by plotnick**
Debugger works in all three host languages. Displays application instances using their native string representations:
```
>>> from oso import Oso, Http
>>> o = Oso()
>>> o.load_str("d(x) if debug(x);")
>>> o.query_predicate("d", Http(hostname="foo")).success
Welcome to the debugger!
debug(Http(hostname="foo"))
debug> bindings
Http = <class 'oso.extras.Http'>
Datetime = <class 'datetime.datetime'>
Timedelta = <class 'datetime.timedelta'>
PathMapper = <class 'oso.extras.PathMapper'>
_x_1 = Http(hostname="foo")
_x_2 = Http(hostname="foo")
debug> continue
True
```

PR checklist:
- [x] Docs

### CODE REVIEW

File: polar/src/formatting.rs [hunk 1]
```diff
@@ -299,7 +299,11 @@ pub mod to_polar {
 
     impl ToPolarString for ExternalInstance {
         fn to_polar(&self) -> String {
-            format!("^{{id: {}}}", self.instance_id)
+            if let Some(ref repr) = self.repr {
+                repr.clone()
+            } else {
+                format!("^{{id: {}}}", self.instance_id)
```
Comment (samscott89) [inline polar/src/formatting.rs:305]: Hmm, actually not sure this should go in ToPolarString... It's not quite correct. ToPolarString should be parseable back into a term.

**Comment by samscott89** [comment]
Reminded by @gj we need docs + tests for these. I think the docs are actually stable enough now that these don't _need_ to be done on the `docs/` branch. But I'd be okay with doing tests here + docs there if easier.

File: polar/src/formatting.rs (hunk 1 above)
Comment (plotnick) [reply polar/src/formatting.rs:305]: `ExternalInstance` was and is the exception to that. There is no parseable notation for an external instance.

Comment (samscott89) [reply polar/src/formatting.rs:305]: Oh you're totally right, I thought we had parser code still for `^{...}`. I think this is still unfortunate though. Other than implementing a _new_ format trait which uses ToPolarString for most things, I can't see a good way to solve this. We should revisit this though.

Comment (plotnick) [reply polar/src/formatting.rs:305]: I don't think it's a problem. E.g., in Python, the default instance repr is not parsable; Common Lisp has `print-unreadable-object` for the same reason. Most instances don't have a parsable representation in most languages.

**Pull request merged**